### DISCUSSION TITLE
LSP mode with pyright on Windows using network drive fails to initialize

### DISCUSSION THREAD

**Discussion by garyo** [emacs-lsp/lsp-mode#3481] [General]
I have emacs 29.0.50, running latest lsp-mode and pyright as my python language server. This works fine as long as the project is local, on my C drive. If I have a project on a network drive (`q:/DEV/gober/toolkit/foo.py`) then the language server fails to start. Here's the verbose log (lightly edited):
```
[Trace - 10:46:11 AM] Sending request 'initialize - (1)'.
Params: {
  "processId": null,
  "rootPath": "q:/DEV/gober/toolkit",
  "clientInfo": {
    "name": "emacs",
    "version": "GNU Emacs 29.0.50 (build 1, x86_64-w64-mingw32)"
  },
  "rootUri": "file:///q%3A/DEV/gober/toolkit",
  "capabilities": {
    "workspace": {
      "workspaceEdit": {
        "documentChanges": true,
        "resourceOperations": [
          "create",
          "rename",
          "delete"
        ]
      },
...
  "initializationOptions": null,
  "workDoneToken": "1",
  "workspaceFolders": [
    {
      "uri": "file:///q%3A/DEV/gober/toolkit",
      "name": "tk-default-config"
    },
    {
      "uri": "file://///fileserver/topdir/DEV/gober/toolkit",
      "name": "tk-default-config"
    }
  ]
}


[Trace - 10:46:13 AM] Received notification 'window/logMessage'.
Params: {
  "type": 3,
  "message": "Pyright language server 1.1.239 starting"
}


[Trace - 10:46:13 AM] Received notification 'window/logMessage'.
Params: {
  "type": 3,
  "message": "Server root directory: C:\\Users\\gober\\AppData\\Roaming\\nvm\\v18.0.0\\node_modules\\pyright/dist/"
}


[Trace - 10:46:13 AM] Received response 'initialize - (1)' in 311ms.
Result: {
  "code": -32603,
  "message": "Request initialize failed with message: [UriError]: If a URI does not contain an authority component, then the path cannot begin with two slash characters (\"//\")"
}
```
I presume the error is coming from the second `workspaceFolders` with ` "file://///fileserver/topdir/DEV/gober/toolkit"`. 

How can I debug this and hopefully fix it? I'm not sure what is doing the path normalization to UNC from drive-letter, or why there are two `workspaceFolders` that are actually the same. I'm happy to dig in and start using `edebug` to track things down, just not sure where to look.

**Comment by yyoncho** [comment]
What is the `(buffer-file-name)` in that directory?

**Comment by garyo** [comment]
`"q:/DEV/gober/toolkit/tk-default-config/hooks/tk-multi-publish2/maya/collector.py"`

**Reply by yyoncho** [reply]
TBO I dont quite understand what is happening. To me, it looks like `file://///fileserver/topdir/DEV/gober/toolkit` and `file:///q%3A/DEV/gober/toolkit` are the same thing, and do you actually need the second? If no, then use M-x lsp-workspace-folders-remove to delete it.

**Comment by garyo** [comment]
Hey, that worked! Will that persist across sessions?
Now my only question is how it got to be that way, but at least I've got it working now.

**Reply by yyoncho** [reply]
It will. I guess by mistake somehow you have added the wrong folder (most likely opening the file using a different path from emacs perspective).
